OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -344,7 +344,13 @@ private static void HandleSuffixPronoun(ref StringBuilder formattedString, ReadO
                         continue;
                     case StringFormatEncoder.FormatSuffix.OrdinalIndicator:
                         // TODO: if the preceding expression value is not a float, it should be replaced with 0 (0th)
-                        if (interps[prevInterpIndex].TryGetValueAsFloat(out var ordinalNumber)) {
+                        if (interps[prevInterpIndex].TryGetValueAsInteger(out var ordinalNumber)) {
+
+                            // For some mystical reason byond converts \th to integers
+                            // This is slightly hacky but the only reliable way I know how to replace the number
+                            // Need to call stringy to make sure its the right length to cut
+                            formattedString.Length -= interps[prevInterpIndex].Stringify().Length;
```

Comment:
pretty sure stringify is the correct thing to do here but im not sure. we'll probably find out soon enough